src/modules/expenses/expense.routes.ts

```diff
@@ -0,0 +1,87 @@
+import { createRoute } from "@hono/zod-openapi";
+import { z } from "zod";
+
+import jsonContentRequired from "@/common/helpers/json-content-required.helper";
+import { jsonContent } from "@/common/helpers/json-content.helper";
+import {
+  authMiddleware,
+  requireAuth,
+} from "@/common/middlewares/auth.middleware";
+import createErrorSchema from "@/common/schema/create-error.schema";
+import { AUTHORIZATION_ERROR_MESSAGE, INTERNAL_SERVER_ERROR_MESSAGE, VALIDATION_ERROR_MESSAGE } from "@/common/utils/constants";
+import * as HTTPStatusCodes from "@/common/utils/http-status-codes.util";
+import {
+  insertExpenseSchema,
+  selectExpenseSchema,
+} from "@/db/schemas/expense.model";
+
+const tags = ["Expenses"];
+
+export const createExpenseRoute = createRoute({
+  tags,
+  method: "post",
+  path: "/expenses",
+  middleware: [
+    authMiddleware(),
+    requireAuth(),
+  ] as const,
+  request: {
+    body: jsonContentRequired(
+      insertExpenseSchema
+        .omit({
+          id: true,
+          createdAt: true,
+          updatedAt: true,
+          creatorId: true,
+        })
+        .partial({
+          payerId: true,
+        }),
+      "Expense creation details",
+    ),
+  },
+  responses: {
+    [HTTPStatusCodes.CREATED]: jsonContent(
+      z.object({
+        success: z.boolean().default(true),
+        message: z.string(),
+        data: selectExpenseSchema,
+      }),
+      "Expense created successfully",
+    ),
+    [HTTPStatusCodes.BAD_REQUEST]: jsonContent(
+      z.object({
+        success: z.boolean().default(false),
+        message: z.string(),
+      }),
+      VALIDATION_ERROR_MESSAGE,
+    ),
+    [HTTPStatusCodes.UNAUTHORIZED]: jsonContent(
+      z.object({
+        success: z.boolean().default(false),
+        message: z.string(),
+      }),
+      AUTHORIZATION_ERROR_MESSAGE,
+    ),
+    [HTTPStatusCodes.UNPROCESSABLE_ENTITY]: jsonContent(
+      createErrorSchema(insertExpenseSchema),
+      VALIDATION_ERROR_MESSAGE,
+    ),
+    [HTTPStatusCodes.NOT_FOUND]: jsonContent(
+      z.object({
+        success: z.boolean().default(false),
+        message: z.string(),
+      }),
+      VALIDATION_ERROR_MESSAGE,
```

Comment:
Not found should not be same as validation error, please check.

Reply:
@shivamvijaywargi i updated error message to `Not found error(s)`. Works?

<img width="431" alt="image" src="https://github.com/user-attachments/assets/28ffa36d-5983-432e-85d7-93abff686b40">


And thank you for getting my attention to "Not Found" status, with this i got to know some mistakes in test cases related to 404. I have fixed them as well.  https://github.com/GenieWizards/finance-management-api/pull/68/commits/5ae08b5612c325f84f467dd94e32dfb4b4f3b641

Reply:
How about "Requested resource not found"?

Reply:
Ya this is good. I will update it.